community/values/src/main/java/org/neo4j/values/BooleanArray.java

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright (c) 2002-2017 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.values;
+
+import java.util.Arrays;
+
+import static java.lang.String.format;
+
+abstract class BooleanArray extends ArrayValue
+{
+    abstract boolean[] value();
+
+    @Override
+    public int length()
+    {
+        return value().length;
+    }
+
+    public boolean booleanValue( int offset )
```

Comment:
This is present for symmetry with `IntergralArray`, `FloatingPointArray` and `TextArray`, and is used by `NumberValues.compareBooleanArrays()`. It does not override because `BooleanArray` happens to be the only representation of boolean array values, while the other 3 types have multiple representations.